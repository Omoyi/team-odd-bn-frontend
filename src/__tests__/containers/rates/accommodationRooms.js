import React from 'react';
import { shallow } from 'enzyme';
import {
  AccommodationsRooms,
  mapStateToProps,
} from '../../../views/accommodation/AccommodationsRooms';

import {
  mapStateToProspReviews,
  accommodationRoomsProps,
  roomsImagesPrevProps,
} from '../../../__mocks__/RatingMocks/BookingAccommodation';


describe('Expect to render all rooms', () => {
  const wrapper = shallow(<AccommodationsRooms {...accommodationRoomsProps} />);
  it('Should retrieve all roooms images successfully', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Should update new image when new accommodation is displayed', () => {
    const updateItem = jest.spyOn(wrapper.instance(), 'componentDidUpdate');
    wrapper.instance().componentDidUpdate(roomsImagesPrevProps.prevProps);
    expect(updateItem).toHaveBeenCalledWith(roomsImagesPrevProps.prevProps);
  });
  it('Should returns items with map state to props', () => {
    const result = mapStateToProps(mapStateToProspReviews.state);
    expect(result).toEqual({ accommodationItem: { rates: 'accommodation items follows' } });
  });
});
