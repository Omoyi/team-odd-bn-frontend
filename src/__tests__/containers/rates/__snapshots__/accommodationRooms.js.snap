// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render all rooms Should retrieve all roooms images successfully 1`] = `ShallowWrapper {}`;
